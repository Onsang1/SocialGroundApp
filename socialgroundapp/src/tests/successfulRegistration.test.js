/**
* @jest-environment jsdom
*/

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import { expect, test } from '@jest/globals';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import SuccessfulRegistration from '../components/successfulRegistration';

test('renders successful page', () => {
  render(
    <Router>
      <SuccessfulRegistration />
    </Router>,
  );
  const linkElement = screen.getByText('Click here');
  expect(linkElement).toBeInTheDocument();
});

test('successfulRegistration page matches snapshot', () => {
  render(
    <Router>
      <SuccessfulRegistration />
    </Router>,
  );
  const component = renderer.create(<SuccessfulRegistration />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
