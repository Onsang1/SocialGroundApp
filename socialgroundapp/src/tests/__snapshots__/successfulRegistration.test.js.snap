// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`successfulRegistration page matches snapshot 1`] = `
<div
  id="react-root"
>
  <div
    className="successfulRegistration"
  >
    <div
      className="small-logo-redirect"
    >
      <a
        href="/"
      >
        <div
          aria-disabled="false"
          role="button"
        >
          <img
            alt="small_logo"
            className="small_logo"
            src={Object {}}
          />
        </div>
      </a>
    </div>
    <div
      className="successfulMessage"
      id="successfulMessage"
    >
      <div
        className="successfulMessageRow"
      >
        Your account has been successfully created! You will be redirected to the Login Page.
      </div>
      <div
        className="successfulMessageRow"
      >
        <p
          className="mainPageredirect"
        >
          <a
            href="/"
          >
            <span
              className="redirectClick"
            >
              Click here
            </span>
          </a>
          Â if the page is not redirected after a while.
        </p>
      </div>
    </div>
  </div>
</div>
`;
